src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```
@@ -1871,6 +1872,37 @@ because xbuild doesn't support framework reference assemblies.
 <Target Name="_LinkAssemblies"
   DependsOnTargets="_ResolveAssemblies;_CreatePackageWorkspace;$(_BeforeLinkAssemblies);_GenerateJniMarshalMethods;_LinkAssembliesNoShrink;_LinkAssembliesShrink" />
 
+<Target Name="_PrepareNativeAssemblySources">
+  <PrepareAbiItems
+    BuildTargetAbis="@(_BuildTargetAbis)"
+    ItemNamePattern="$(_NativeAssemblySourceDir)typemaps.@abi@.s">
+      <Output TaskParameter="OutputItems" ItemName="_TypeMapAssemblySource" />
+  </PrepareAbiItems>
+
+</Target>
+<Target Name="_GenerateTypeMaps"
+    DependsOnTargets="_PrepareNativeAssemblySources"
+    Inputs="@(ResolvedAssemblies);@(ResolvedUserAssemblies)"
```

It may be easier to move this to the `_GenerateJavaStubs` target.

If not, I think these `Inputs` should more closely match what `_GenerateJavaStubs` has:
```suggestion
    Inputs="$(MSBuildAllProjects);@(_ResolvedUserMonoAndroidAssemblies);$(_AndroidBuildPropertiesCache)"
```

---

I split it out of `GenerateJavaStubs` on purpose - to make it easier to modify and maintain (and I thought `GenerateJavaStubs` was to be split up to smaller bits)

---

I, kind of, merged the code back into `GenerateJavaStubs`. I kept the `GenerateTypeMaps` task nearly as is and I call it from `GenerateJavaStubs`. I also left the task invocation in the targets file, but dependency on it is commented out so it never runs. This way it will be easy to refactor it when someone gets to splitting `GenerateJavaStubs` up